### DISCUSSION TITLE
How to include Adafruit_GFX_Button in to Arduino_GFX ?

### DISCUSSION THREAD

**Discussion by Vint232** [moononournation/Arduino_GFX#221] [General]
I tried to find something similar to tft_espi button. but i didn't find anything.
I want to test how Arduino_GFX works with Adafruit_GFX_Button or tft_espi button. Maybe you can install Adafruit_GFX_Button into Arduino_GFX library?

**Comment by moononournation** [comment]
If you would like to have some touch UI interface, try LVGL examples.

**Reply by Vint232** [reply]
yes lvgl. good graphical interface but it's windows 11. i don't need it. I want to make a simple interface. on ESP32_4827S043. I did it on tft_espi 3.5 lcd.
![IMG_2098](https://user-images.githubusercontent.com/80031860/200755298-4e4b0dff-b4d0-4343-88e5-29b0041a344d.jpg)

**Reply by moononournation** [reply]
Have you tried this example?

examples/LVGL/LvglWidgets

**Reply by Vint232** [reply]
Yes I try it. But I don't like. Becouse I cant make what I want. LVGL made more for home user interface. I want to make a simple  service inteface. LVGL don't have serial terminall since ver 5. 
I'm not so smart that I could do it myself. I use more ready-made solutions for my needs.

**Reply by Vint232** [reply]
can you find where the error is? I can't get it to compile
[HelloWorld_OK_06b.zip](https://github.com/moononournation/Arduino_GFX/files/9973293/HelloWorld_OK_06b.zip)

**Comment by TangerineDreamer** [comment]
i'm also much interested making buttons like tft_espi with gfx,how can i?
LovyanGFX allow this :  LGFX_Button key[15]; to reproduce tft_espi button generator,what about GFX?

**Reply by moononournation** [reply]
I am encouraging everyone can build their own UI base on top of GFX library. But the function drawing the high level UI objects should not part of GFX library. Because what button should look like is very subjective, it cannot implement a single draw button function that can fulfill everyone's beautiful mind. If you do not have time to make one yourself and want some pre-made UI, I recommend you try LVGL.

**Comment by TangerineDreamer** [comment]
Hi, moononournation,
We don't want to overload you with such a task, i think you sure know this function : LGFX_Button key[]
it consist of building a multi button keypad using arrays for names and colors,avoiding the repetition of
creating xtimes the same button but at a different coordonates,creating a button is not difficult to do,but
this function is very handy and could be useful for your arduino_ gfx :)
Arduino_gfx is the recommended library by the sellers (of ESP32S3_RGB_8048S070C , I own),but lvgl is too complicate for basic users like me,i would like to use tft_espi with the button generator too,but it's not supported yet,maybe never.

**Reply by moononournation** [reply]
May be using TFT_eSPI with button generator is a good option for you.

**Comment by TangerineDreamer** [comment]
of course but not working with my ESP32S3_RGB_8048S070C ,not supported by tft_espi :)

https://www.makerfabs.com/sunton-esp32-s3-7-inch-tn-display-with-touch.html

only arduino_gfx work with,that's why i asked you
